src/Core/maps/src/Handlers/Map/MapHandler.Android.cs

```diff
@@ -0,0 +1,632 @@
+
+				if (coarseLocationPermission == Permission.Granted || fineLocationPermission == Permission.Granted)
+				{
+					googleMap.MyLocationEnabled = googleMap.UiSettings.MyLocationButtonEnabled = true;
+				}
+				else
+				{
+					handler.MauiContext?.Services.GetService<ILogger<MapHandler>>()?.LogWarning("Missing location permissions for IsShowingUser");
+					googleMap.MyLocationEnabled = googleMap.UiSettings.MyLocationButtonEnabled = false;
+				}
+			}
+			else
+			{
+				googleMap.MyLocationEnabled = googleMap.UiSettings.MyLocationButtonEnabled = false;
+			}
+		}
+
+		public static void MapHasScrollEnabled(IMapHandler handler, IMap map)
+		{
+			GoogleMap? googleMap = handler?.Map;
+			if (googleMap == null)
+				return;
+
+			googleMap.UiSettings.ScrollGesturesEnabled = map.HasScrollEnabled;
+		}
+
+		public static void MapHasTrafficEnabled(IMapHandler handler, IMap map)
+		{
+			GoogleMap? googleMap = handler?.Map;
+			if (googleMap == null)
+				return;
+
+			googleMap.TrafficEnabled = map.HasTrafficEnabled;
+		}
+
+		public static void MapHasZoomEnabled(IMapHandler handler, IMap map)
+		{
+			GoogleMap? googleMap = handler?.Map;
+			if (googleMap == null)
+				return;
+
+			googleMap.UiSettings.ZoomControlsEnabled = map.HasZoomEnabled;
+			googleMap.UiSettings.ZoomGesturesEnabled = map.HasZoomEnabled;
+		}
+
+		public static void MapMoveToRegion(IMapHandler handler, IMap map, object? arg)
+		{
+			MapSpan? newRegion = arg as MapSpan;
+			if (newRegion != null)
+				(handler as MapHandler)?.MoveToRegion(newRegion, true);
+		}
+
+		public void UpdateMapElement(IMapElement element)
+		{
+			switch (element)
+			{
+				case IGeoPathMapElement polyline:
+					{
+						if (element is IFilledMapElement polygon)
+						{
+							PolygonOnPropertyChanged(polyline);
+						}
+						else
+						{
+							PolylineOnPropertyChanged(polyline);
+						}
+						break;
+					}
+				case ICircleMapElement circle:
+					{
+						CircleOnPropertyChanged(circle);
+						break;
+					}
+			}
+		}
+
+		void PolygonOnPropertyChanged(IGeoPathMapElement formsPolygon)
+		{
+			var nativePolygon = GetNativePolygon(formsPolygon);
+
+			if (nativePolygon == null)
+				return;
+
+			if (formsPolygon.Stroke is SolidPaint solidPaint)
+				nativePolygon.StrokeColor = solidPaint.Color.AsColor();
+
+			if ((formsPolygon as IFilledMapElement)?.Fill is SolidPaint solidFillPaint)
+				nativePolygon.FillColor = solidFillPaint.Color.AsColor();
+
+			nativePolygon.StrokeWidth = (float)formsPolygon.StrokeThickness;
+			nativePolygon.Points = formsPolygon.Select(position => new LatLng(position.Latitude, position.Longitude)).ToList();
+		}
+
+		void PolylineOnPropertyChanged(IGeoPathMapElement formsPolyline)
+		{
+			var nativePolyline = GetNativePolyline(formsPolyline);
+
+			if (nativePolyline == null)
+				return;
+
+			if (formsPolyline.Stroke is SolidPaint solidPaint)
```

Comment:
It's just a detail, but should we avoid the words forms, xamarinforms, etc and use virtualview, platform etc instead? Just having in mind possible community contributors who don't have Xamarin.Forms context and avoid confusion by using VirtualView and PlatformView everywhere possible.

Reply:
Sure